From e8edeb9a468a8307c3e3acdf26d281e749df24e9 Mon Sep 17 00:00:00 2001
From: tstegema <tstegema@90b65887-3827-0410-9a23-83215b262276>
Date: Thu, 24 Jan 2019 17:59:27 +0000
Subject: [PATCH 0943/1076] MANTIS 0053039 SteT: Support AZTEC barcode printing
 for LH

massai_graphic.lib
V1.0.0.8
- allow to set columns for pdf417


git-svn-id: svn://localhost/SelfServiceCommon/trunk@1429 90b65887-3827-0410-9a23-83215b262276
---
 SelfServiceCommon/Interfaces/inc/graphic/barcodePDF417.h | 3 ++-
 .../Massai/cpp/Graphic/src/TestBarcodePDF417.cpp         | 5 +++--
 .../Massai/cpp/Graphic/src/barcodePDF417.cpp             | 8 ++++++--
 SelfServiceCommon/Massai/cpp/Graphic/src/local_def.cpp   | 9 +++++++--
 4 files changed, 18 insertions(+), 7 deletions(-)

diff --git a/SelfServiceCommon/Interfaces/inc/graphic/barcodePDF417.h b/SelfServiceCommon/Interfaces/inc/graphic/barcodePDF417.h
index a5303965..e78a47e4 100644
--- a/SelfServiceCommon/Interfaces/inc/graphic/barcodePDF417.h
+++ b/SelfServiceCommon/Interfaces/inc/graphic/barcodePDF417.h
@@ -15,7 +15,8 @@ namespace MASSAI
     {
       size_t blockWidth;
       size_t blockHeight;
-      size_t rows;
+      size_t rows;        // use rows to estimate columns if columns is not set
+      size_t columns;
       size_t errorLevel;  // 0-8: 2^(errorLevel+1)
     };
 
diff --git a/SelfServiceCommon/Massai/cpp/Graphic/src/TestBarcodePDF417.cpp b/SelfServiceCommon/Massai/cpp/Graphic/src/TestBarcodePDF417.cpp
index 5adfb13d..70fa672c 100644
--- a/SelfServiceCommon/Massai/cpp/Graphic/src/TestBarcodePDF417.cpp
+++ b/SelfServiceCommon/Massai/cpp/Graphic/src/TestBarcodePDF417.cpp
@@ -30,7 +30,7 @@ void printUsage(int argc, char** argv, char* error= 0)
   std::cout << " -out:<BitmapFile>";
   std::cout << " [-width:<BlockWidth>]";
   std::cout << " [-height:<BlockHeight>]";
-  std::cout << " [-rows:<NumberOfRows]";
+  std::cout << " [-columns:<NumberOfColumns]";
   std::cout << " [-error:<ErrorLevel]";
   std::cout << std::endl;
   if(error)
@@ -84,7 +84,8 @@ int main(int argc, char** argv)
       BarcodePDF417Settings settings;
       settings.blockWidth= args.optionInteger("WIDTH", 1);
       settings.blockHeight= args.optionInteger("HEIGHT", 1);
-      settings.rows= args.optionInteger("ROWS", 0);
+      settings.rows = 0;
+      settings.columns= args.optionInteger("COLUMNS", 1);
       settings.errorLevel= args.optionInteger("ERROR", 3);
       createBarcode(filename, input, settings);
     }
diff --git a/SelfServiceCommon/Massai/cpp/Graphic/src/barcodePDF417.cpp b/SelfServiceCommon/Massai/cpp/Graphic/src/barcodePDF417.cpp
index 5cdb2c2a..42cb0a1f 100644
--- a/SelfServiceCommon/Massai/cpp/Graphic/src/barcodePDF417.cpp
+++ b/SelfServiceCommon/Massai/cpp/Graphic/src/barcodePDF417.cpp
@@ -28,8 +28,12 @@ namespace
       barcode->show_hrt = 0;
 
       int codewords = data.size() / 2 + (2 << (settings.errorLevel + 1));
-      int columns = codewords / settings.rows;
-      columns = std::min(30, std::max(1, columns));
+      int columns = settings.columns;
+      if(!columns)
+      {
+        columns = codewords / settings.rows;
+        columns = std::min(30, std::max(1, columns));
+      }
 
       barcode->option_2 = columns;
       barcode->option_1 = settings.errorLevel;
diff --git a/SelfServiceCommon/Massai/cpp/Graphic/src/local_def.cpp b/SelfServiceCommon/Massai/cpp/Graphic/src/local_def.cpp
index 2e56d109..8cfef09a 100644
--- a/SelfServiceCommon/Massai/cpp/Graphic/src/local_def.cpp
+++ b/SelfServiceCommon/Massai/cpp/Graphic/src/local_def.cpp
@@ -1,6 +1,6 @@
 #include "tools/versionlib.h"
 
-VERSION_ID_DT(graphic, 1, 0, 0, 7);
+VERSION_ID_DT(graphic, 1, 0, 0, 8);
 
 /*!
 \file
@@ -8,12 +8,17 @@ VERSION_ID_DT(graphic, 1, 0, 0, 7);
 
 \page massai_graphic.lib Release Notes - massai_graphic.lib
 
-@version 1.0.0.7
+@version 1.0.0.8
 
 \section changehistory Change history
 
 massai_graphic.lib
 
+\version 1.0.0.8
+\li 24.01.2019
+\li SteT bugfix Mantis 0053039: Support AZTEC barcode printing for LH
+- allow to set columns for pdf417
+
 \version 1.0.0.7
 \li 17.01.2019
 \li SteT bugfix Mantis 0053039: Support AZTEC barcode printing for LH
-- 
2.41.0.windows.1

